service/dap/server_test.go

```diff
@@ -2296,6 +2296,108 @@ func TestSetBreakpoint(t *testing.T) {
 	})
 }
 
+// TestSetInstructionBreakpoint executes to a breakpoint and tests different
+// configurations of setInstructionBreakpoint requests.
+func TestSetInstructionBreakpoint(t *testing.T) {
+	runTest(t, "loopprog", func(client *daptest.Client, fixture protest.Fixture) {
+		runDebugSessionWithBPs(t, client, "launch",
+			// Launch
+			func() {
+				client.LaunchRequest("exec", fixture.Path, !stopOnEntry)
+			},
+			// Set breakpoints
+			fixture.Source, []int{16}, // b main.main
+			[]onBreakpoint{{
+				execute: func() {
+					checkStop(t, client, 1, "main.main", 16)
+
+					// Set two breakpoints in the loop
+					client.SetBreakpointsRequest(fixture.Source, []int{8, 9})
+					expectSetBreakpointsResponse(t, client, []Breakpoint{{8, fixture.Source, true, ""}, {9, fixture.Source, true, ""}})
+
+					// Continue to the two breakpoints and get the instructionPointerReference.
+					client.ContinueRequest(1)
+					client.ExpectContinueResponse(t)
+					client.ExpectStoppedEvent(t)
+					checkStop(t, client, 1, "main.loop", 8)
+
+					client.StackTraceRequest(1, 0, 1)
+					st := client.ExpectStackTraceResponse(t)
+					if len(st.Body.StackFrames) < 1 {
+						t.Fatalf("\ngot  %#v\nwant len(stackframes) => 1", st)
+					}
+					pc8 := st.Body.StackFrames[0].InstructionPointerReference
+
+					client.ContinueRequest(1)
+					client.ExpectContinueResponse(t)
+					client.ExpectStoppedEvent(t)
+					checkStop(t, client, 1, "main.loop", 9)
+
+					client.StackTraceRequest(1, 0, 1)
+					st = client.ExpectStackTraceResponse(t)
+					if len(st.Body.StackFrames) < 1 {
+						t.Fatalf("\ngot  %#v\nwant len(stackframes) => 1", st)
+					}
+					pc9 := st.Body.StackFrames[0].InstructionPointerReference
+
+					// Clear the source breakpoints.
+					// TODO(suzmue): there is an existing issue that breakpoints with identical locations
+					// from different setBreakpoints, setFunctionBreakpoints, setInstructionBreakpoints
+					// requests will prevent subsequent ones from being set.
```

Comment:
What do you mean? Does this mean that if a person sets a breakpoint at a location and then changes it to be a different kind of breakpoint, it won't work?

Reply:
Already if a user sets a function breakpoint at name=myFunction which is defined in main.go:17 and a source breakpoint at file=main.go line=17. And lets say the function breakpoint is set first. Then if the function breakpoint is removed, there will no longer be a breakpoint set at main.go:17, since the source breakpoint will have failed to set because there was already a breakpoint there.